WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -121,10 +123,19 @@ public void onMessageReceived(String from, Bundle data) {
             return;
         }
 
-        // Handle helpshift PNs
-        if (TextUtils.equals(data.getString("origin"), "helpshift")) {
-            // TODO: Handle Zendesk PNs instead
-//            HelpshiftHelper.getInstance().handlePush(this, new Intent().putExtras(data));
+        // Handle Zendesk PNs
+        // TODO: Move the contents of this to a helper
+        if (TextUtils.equals(data.getString("type"), "zendesk")) {
+            // TODO: Is it possible to keep the current `WPMainActivity`? App opening animation can be distracting
+            Intent resultIntent = new Intent(this, WPMainActivity.class);
+            resultIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+            resultIntent.putExtra(WPMainActivity.ARG_OPEN_PAGE, WPMainActivity.ARG_ME);
+
+            String title = data.getString(PUSH_ARG_TITLE);
+            String message = data.getString(PUSH_ARG_MSG);
+            // TODO: figure out a proper pushId
+            int pushId = Integer.parseInt(new SimpleDateFormat("ddHHmmss", Locale.US).format(new Date()));
```

Comment:
I was toying with a similar idea in my head, since we don't deep-link the ticket, all push notifications from Zendesk will pretty much be the same. So, using the same id should be just fine.

I'll keep thinking about it to see if I am missing something, but if not I'll go with this. We just need to make sure we don't use an id that might be used for a wpcom notification id.